elements/lisk-dpos/src/delegates_list.ts

```diff
@@ -121,26 +171,165 @@ export const getForgerPublicKeysForRound = async (
+			// Insert only if account is a delegate
+			if (account.username) {
+				updatedAccountsMap[account.address] = account;
+			}
+		}
+		// Inject delegate account if it doesn't exist
+		for (const delegate of originalDelegates) {
+			if (updatedAccountsMap[delegate.address] === undefined) {
+				updatedAccountsMap[delegate.address] = delegate;
+			}
+		}
+		const delegates = [...Object.values(updatedAccountsMap)];
+
+		// Update totalVotesReceived to voteWeight equivalent before sorting
+		for (const account of delegates) {
+			// If the account is being punished, then consider them as vote weight 0
+			if (isCurrentlyPunished(height, account.delegate.pomHeights)) {
+				account.totalVotesReceived = BigInt(0);
+				continue;
+			}
+			const selfVote = account.votes.find(
+				vote => vote.delegateAddress === account.address,
+			);
+			const cappedValue =
+				(selfVote?.amount ?? BigInt(0)) * BigInt(this.voteWeightCapRate);
+			if (account.totalVotesReceived > cappedValue) {
+				account.totalVotesReceived = cappedValue;
+			}
+		}
+
+		delegates.sort((a, b) => {
+			const diff = b.totalVotesReceived - a.totalVotesReceived;
+			if (diff > BigInt(0)) {
+				return 1;
+			}
+			if (diff < BigInt(0)) {
+				return -1;
+			}
+
+			return a.address.localeCompare(b.address, 'en');
+		});
+
+		const activeDelegates = [];
+		const standbyDelegates = [];
+		for (const account of delegates) {
+			// If the account is banned, do not include in the list
+			if (account.delegate.isBanned) {
+				continue;
+			}
+
+			// Select active delegate first
+			if (activeDelegates.length < this.activeDelegates) {
+				activeDelegates.push({
+					address: account.address,
+					voteWeight: account.totalVotesReceived.toString(),
+				});
+				continue;
+			}
+
+			// If account has more than threshold, save it as standby
+			if (account.totalVotesReceived >= this.standbyThreshold) {
+				standbyDelegates.push({
+					address: account.address,
+					voteWeight: account.totalVotesReceived.toString(),
+				});
+				continue;
+			}
+
+			// From here, it's below threshold
+			// Below threshold, but prepared array does not have enough slected delegate
+			if (
+				standbyDelegates.length < this.standbyDelegates
+			) {
+				// In case there was 1 standby delegate who has more than threshold
+				standbyDelegates.push({
+					address: account.address,
+					voteWeight: account.totalVotesReceived.toString(),
+				});
+				continue;
+			}
+			break;
+		}
+
+		const result = activeDelegates.concat(standbyDelegates);
+		const voteWeight = {
+			round,
+			delegates: result,
+		};
+		// Save result to the chain state with round number
+		const voteWeights = await getVoteWeights(stateStore);
+		const voteWeightsIndex = voteWeights.findIndex(vw => vw.round === round);
+		if (voteWeightsIndex > 0) {
+			voteWeights[voteWeightsIndex] = voteWeight;
+		} else {
+			voteWeights.push(voteWeight);
+		}
+		_setVoteWeights(stateStore, voteWeights);
```

Comment:
```suggestion
		const extendedDelegateList = activeDelegates.concat(standbyDelegates);
		const delegateListForARound = {
			round,
			delegates: extendedDelegateList,
		};
		// Save delegateListForARound to the chain state with round number
		const voteWeights = await getVoteWeights(stateStore);
		const voteWeightsIndex = voteWeights.findIndex(vw => vw.round === round);
		if (voteWeightsIndex > 0) {
			voteWeights[voteWeightsIndex] = delegateListForARound;
		} else {
			voteWeights.push(delegateListForARound);
		}
		_setVoteWeights(stateStore, voteWeights);
```

Reply:
This is not delegate list, it is the voteWeights snapshot.  calling it as delegate list will be confusing IMO

Reply:
it captures for a round list if delegates ordered by totalVoteWeight (active + standby), so `voteWeight` sounds more like voteWeight of the round. but internally its the list of delegates for a round order by totalVoteWeight, am i correct?

Reply:
vote weight is the effective vote weight per delegate.
Vote weight considers punishment, self vote cap, ban and standby threshold using `totalVotesReceived`.
But this list is not directly used for forging. It processes one more time to the forgers list.
List itself is called `voteWeights`, so one element is `voteWeight`, but i changed the list name to `delegateVoteWeights`